articles/introduction-to- virtual- environments-and-dependency-managers/index.md

```diff
@@ -0,0 +1,419 @@
+
+```bash
+pipenv --rm
+```
+
+Always make sure to be working in your project directory terminal before using these commands.
+
+**Note:** With Pipenv, you can force install a specific python version for your project using:
+
+```bash
+pipenv install –python 3.6
+```
+
+### Advantages of Pipenv
+
+1. It automatically creates and manages a virtual environment for your projects.
+2. It generates the Pipfile.lock file which helps in deterministic builds.
+3. It adds and removes packages from a Pip file as you install and uninstall packages.
+4. It automatically locate you project location by looking for a Pip file.
+
+### 2. Venv
+
+### What is Venv?
+
+Venv is a Python's module that is used to create light weight virtual environments. It is the standard virtual environment module for **Python 3**.
+It is one of the easiest modules to use for creating virtual environments.
+
+### The problem it solves
+
+* Installation of python packages and libraries of different versions in the same computer.
+
+### Installation
+
+Since Venv comes pre-installed in Python 3 and above, there's no need for installing it.
+
+So, we will go ahead and create a virtual environment.
+
+### Creating a virtual environment
+
+when working on windows Command Prompt, use:
+
+```bash
+> python3 -m venv \path_to\my_project_env
+```
+
+### Activating the virtual environment
+
+Before using the virtual environment, we need to activate it. This makes the current directory you're working on to temporarily function as the default python interpreter.
+
+Activate your project's environment by using the following command:
+
+```bash
+> cd \path_to\my_project\venv\Scripts>activate.bat
+```
+
+On Windows PowerShell, use:
+
+```bash
+path_to\my_project\venv\Scripts>Activate.ps1
+```
+
+ You've successfully activated your environment once the path changes to :
+
+```bash
+(venv) path_to\my_project\venv\Scripts>
+```
+
+### Configuring the virtual environment
+
+After activating the virtual environment, you can now start installing modules and packages by using the great *`pip install`* command.
+
+For example:
+
+```bash
+pip install Django
+```
+
+Check the version of Pip you're using and directory with the following command:
+
+```bash
+(venv) path_to\my_project\venv\Scripts> pip -V
+```
+
+>**Note**: It is a good practice to always work with a *requirements.txt* file for easy collaboration since it stores a list of all installed packages in your environment.
+In order to create one, use
+
+``` bash
+(venv) path_to\my_project\venv\Scripts> python -m pip freeze > requirements.txt` 
+```
+
+or manually create the file and add the packages installed.
+
+### Deactivating
+
+After you finish with the environment, deactivate it using:
+
+```bash
+(venv) path_to\my_project\venv\Scripts>deactivate.bat
+```
+
+### Deleting the virtual environment
+
+Since virtual environments are self-contained, when no longer needed, just delete the project's directory.
```

Comment:
I guess you wanted to say virtual environment directory.
```suggestion
Since virtual environments are self-contained, when no longer needed, just delete the virtual environment directory.
```